code/obj/item/kitchen.dm

```diff
@@ -1005,30 +1013,26 @@ TRAYS
 		else if(istype(weapon,/obj/item/platestack))
 			var/obj/item/platestack/p = weapon
 			var/keeptrigger = 0
-			if(((src.platenum + (p.platenum+1)) > 7) && (src.platenum != 7))
+			if(((src.platenum + (p.platenum+1)) > platemax) && (src.platenum != platemax))
 				keeptrigger = 1
-				p.platenum = (p.platenum - (7 - src.platenum))
-				p.icon_state = "platestack[p.platenum]"
-				p.item_state = "platestack[p.platenum]"
-				src.platenum = 7
-				src.icon_state = "platestack[src.platenum]"
-				src.item_state = "platestack[src.platenum]"
-			else if(src.platenum == 7)
+				p.platenum = (p.platenum - (platemax - src.platenum))
+				p.update_icon(user)
+				src.platenum = platemax
+				src.update_icon(user)
+			else if(src.platenum == platemax)
 				boutput(user,"<span style=\"color:red\"><b>The plates are piled too high!</b></span>")
 				return
 			else
 				src.platenum += (p.platenum+1)
-				src.icon_state = "platestack[src.platenum]"
-				src.item_state = "platestack[src.platenum]"
-			if(keeptrigger != 1)
+				src.update_icon(user)
+			if(keeptrigger != platemax)
```

Comment:
sorry, this is where I meant to comment - this will have some FUNK behaviour I think

Reply:
oh fuck
im not sure how i did that, heck

Reply:
fixed in latest push